ptvsd/wrapper.py

```diff
@@ -418,7 +418,12 @@ def make_packet(self, cmd_id, args):
         with self.lock:
             seq = self.seq
             self.seq += 1
-        s = u'{}\t{}\t{}\n'.format(cmd_id, seq, args)
+        try:
+            s = u'{}\t{}\t{}\n'.format(cmd_id, seq, args)
+        except UnicodeDecodeError:
+            reload(sys)
```

Comment:
This isn't the correct fix... rather, you should make sure that you're not mixing bytes and unicode doing the proper encode/decode calls (you have to be careful because it has to work on py2 and 3 so, this should definitely have a test to make sure it works well on both).

Also, making `reload(sys)` is full of caveats and is definitively not something that the debugger should be doing.

If I can suggest, take a look at https://nedbatchelder.com/text/unipain.html for some explanation on how unicode works.

Reply:
@fabioz thank you, I will update the PR likewise